pgx-utils/src/lib.rs

```diff
@@ -29,6 +29,7 @@ pub mod __reexports {
 
 #[derive(Debug, Hash, Eq, PartialEq, Clone, PartialOrd, Ord)]
 pub enum ExternArgs {
+    CreateOrReplace,
```

Comment:
I don't like that this is a single enum, rather than a record type with a bunch of overlapping meanings. For example, `IMMUTABLE`, `STABLE, and `VOLATILE` are mutually exclusive, but there's nothing (in the Rust code) stopping one from specifying both, and getting a crash out of Postgres. Could we re-factor this to something like:
```rust
pub enum FunctionStability { Immutable, Strict, Volatile };
pub enum ParallelSafety { Safe, Unsafe, Restricted };
// ...
pub struct PgFunctionDecorations {
  create_or_replace: bool,
  stability: FunctionStability,
  leakproof: bool,
  ... // cost, no_guard, etc etc
}
```
Then we get more compile-time guarantees that what we generate is going to be acceptable to Postgres.

Reply:
Hm. Do you mean a crash or merely a failure to commit the transaction as Postgres rejects it?

Reply:
This would presumably manifest as a failure for the extension to be installed. The work in #665 should at least make this condition easier to detect, but there's no reason we should be able to generate `CREATE FUNCTION ... IMMUTABLE VOLATILE ...` at all.

Reply:
Mm, I agree, but I am unenthused at the prospect of a bunch of booleans scurrying around.